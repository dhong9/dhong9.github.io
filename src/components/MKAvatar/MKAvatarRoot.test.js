/**
=========================================================
* Danyo-1.2
=========================================================

* Copyright 2024 Danyo (https://www.danyo.tech)

Coded by www.danyo.tech

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
*/

// React testing libraries
import { render } from "@testing-library/react";

// Material Kit 2 React themes
import theme from "assets/theme";

// Component to test
import MKAvatarRoot from "components/MKAvatar/MKAvatarRoot";

describe("MKAvatar", () => {
  it("renders extra small transparent avatars", () => {
    const ownerState = {
      bgColor: "transparent",
      size: "xs",
    };

    const { container } = render(<MKAvatarRoot theme={theme} ownerState={ownerState} />);

    expect(container).toMatchSnapshot();
  });

  it("renders small primary avatars", () => {
    const ownerState = {
      bgColor: "primary",
      size: "sm",
    };

    const { container } = render(<MKAvatarRoot theme={theme} ownerState={ownerState} />);

    expect(container).toMatchSnapshot();
  });

  it("renders large secondary avatars", () => {
    const ownerState = {
      bgColor: "secondary",
      size: "lg",
    };

    const { container } = render(<MKAvatarRoot theme={theme} ownerState={ownerState} />);

    expect(container).toMatchSnapshot();
  });

  it("renders extra large info avatars", () => {
    const ownerState = {
      bgColor: "info",
      size: "xl",
    };

    const { container } = render(<MKAvatarRoot theme={theme} ownerState={ownerState} />);

    expect(container).toMatchSnapshot();
  });

  it("renders default success avatars", () => {
    const ownerState = {
      bgColor: "success",
      size: "ddh",
    };

    const { container } = render(<MKAvatarRoot theme={theme} ownerState={ownerState} />);

    expect(container).toMatchSnapshot();
  });
});
