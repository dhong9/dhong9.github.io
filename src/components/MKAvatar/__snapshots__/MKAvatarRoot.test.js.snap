// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MKAvatar renders default success avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-i2cqlx-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders extra large info avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2ttp3q-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders extra small transparent avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-16u0vg-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders large secondary avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1h6jf0p-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;

exports[`MKAvatar renders small primary avatars 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-c8x9hf-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;
